controllers/configuration_controller_test.go

```diff
@@ -1639,6 +1639,118 @@ func TestAssembleTerraformJobWithGitCredentialsSecretRef(t *testing.T) {
 	assert.Contains(t, spec.Volumes, gitAuthSecretVolume)
 }
 
+func TestAssembleTerraformJobWithTerraformRegistryConfigAndCredentialsSecretRef(t *testing.T) {
+	meta := &TFConfigurationMeta{
+		Name:                "a",
+		ConfigurationCMName: "b",
+		BusyboxImage:        "c",
+		GitImage:            "d",
+		Namespace:           "e",
+		TerraformImage:      "f",
+		RemoteGit:           "g",
+		TerraformRCConfigMapReference: &corev1.SecretReference{
+			Namespace: "default",
+			Name:      "terraform-registry-config",
+		},
+		TerraformCredentialsSecretReference: &corev1.SecretReference{
+			Namespace: "default",
+			Name:      "terraform-credentials",
+		},
+	}
+
+	job := meta.assembleTerraformJob(TerraformApply)
+	spec := job.Spec.Template.Spec
+
+	var terraformSecretDefaultMode int32 = 0400
+
+	terraformRegistryConfigMapVolume := corev1.Volume{Name: TerraformRCConfigVolumeName}
+	terraformRegistryConfigMapVolume.ConfigMap = &corev1.ConfigMapVolumeSource{
+		LocalObjectReference: corev1.LocalObjectReference{
+			Name: "terraform-registry-config",
+		},
+		DefaultMode: &terraformSecretDefaultMode,
+	}
+
+	terraformRegistryConfigVolumeMount := corev1.VolumeMount{
+		Name:      TerraformRCConfigVolumeName,
+		MountPath: TerraformRCConfigVolumeMountPath,
+	}
+	terraformCredentialsSecretVolume := corev1.Volume{Name: TerraformCredentialsConfigVolumeName}
+	terraformCredentialsSecretVolume.Secret = &corev1.SecretVolumeSource{
+		SecretName:  "terraform-credentials",
+		DefaultMode: &terraformSecretDefaultMode,
+	}
+
+	terraformCredentialsSecretVolumeMount := corev1.VolumeMount{
+		Name:      TerraformCredentialsConfigVolumeName,
+		MountPath: TerraformCredentialsConfigVolumeMountPath,
+	}
+
+	assert.Contains(t, spec.InitContainers[0].VolumeMounts, terraformRegistryConfigVolumeMount)
+	assert.Contains(t, spec.Volumes, terraformRegistryConfigMapVolume)
+
+	assert.Contains(t, spec.InitContainers[0].VolumeMounts, terraformCredentialsSecretVolumeMount)
+	assert.Contains(t, spec.Volumes, terraformCredentialsSecretVolume)
+
+}
+
+func TestAssembleTerraformJobWithTerraformRegistryConfigAndCredentialsHelperConfigMapRef(t *testing.T) {
```

Comment:
Why not merge this with the last test?

Reply:
the terraform credentials and the credentials helper are such that only one of them can be used with terraformrc for authentication. Hence have added two test cases to test the different scenarios.

Below are the two scenarios:
1. .terraformrc with credentials.tfrc.json
2. .terraformrc with credentials-helper.

Reply:
merged the two tests.